server/index-node/src/resolver.rs

```diff
@@ -158,6 +159,92 @@ where
         })
     }
 
+    fn resolve_cached_ethereum_calls(
+        &self,
+        field: &a::Field,
+    ) -> Result<r::Value, QueryExecutionError> {
+        let network = field
+            .get_required::<String>("network")
+            .expect("Valid network required");
+
+        let block_hash = field
+            .get_required::<H256>("blockHash")
+            .expect("Valid blockHash required");
+
+        let chain = if let Ok(c) = self
+            .blockchain_map
+            .get::<graph_chain_ethereum::Chain>(network.clone())
+        {
+            c
+        } else {
+            error!(
+                self.logger,
+                "Failed to fetch cached Ethereum calls; nonexistant network";
+                "network" => network,
+                "block_hash" => format!("{}", block_hash),
+            );
+            return Ok(r::Value::Null);
+        };
+        let chain_store = chain.chain_store();
+        let call_cache = chain.call_cache();
+
+        let block_number = match chain_store.block_number(block_hash) {
+            Ok(Some((_, n))) => n,
+            Ok(None) => {
+                error!(
+                    self.logger,
+                    "Failed to fetch cached Ethereum calls; block not found";
+                    "network" => network,
+                    "block_hash" => format!("{}", block_hash),
+                );
+                return Ok(r::Value::Null);
+            }
+            Err(e) => {
+                error!(
+                    self.logger,
+                    "Failed to fetch cached Ethereum calls; storage error";
+                    "network" => network.as_str(),
+                    "block_hash" => format!("{}", block_hash),
+                    "error" => e.to_string(),
+                );
```

Comment:
It would be better to return all these `error!` logs as `QueryExecutionError` in some form; that way, the response to these requests will contain the error, which is much more convenient for troubleshooting (e.g., to figure out why your lookup on the network `mainet` doesn't return anything)

Reply:
Good point, fixed in https://github.com/graphprotocol/graph-node/pull/3343/commits/31a121441011358fe7b202250554df8f060291f2 as well.